Lib/fontTools/designspaceLib/convert5to4.py

```diff
@@ -0,0 +1,538 @@
+# pyright: basic
+
+# FIXME: we're building a stylespace as in statmake but it's not part of fontTools
+# TODO: either move this converter out of fontTools so it can keep creating the stylespace
+#       or make this converter use the fontTools APIs for stat/generate data for these
+# TODO: The STAT table has a fallback name id for styles where everything is elided, but it
+#       is a name id, which exists in a VF, but not in a Designspace. Make accessible somehow?
```

Comment:
maybe this elided fallback name could be yet another attribute in the DS5 _somewhere_

Reply:
I've added the name as a property on DesignSpaceDocument. I have a new question now (@madig):

https://github.com/fonttools/fonttools/blob/97a1bda5c8bcb9999587c24147aeb39b38e7adb5/Lib/fontTools/designspaceLib/convert5to4.py#L75-L80